File: hw/tipan/U3/Hw37Strategy/StrategyPatter/src/main/java/ec/edu/espe/strategypatter/model/BubbleSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategypatter.model;

import java.util.Arrays;

/**
 *
 * @author Lisbeth Tipan
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int[] data) {
         int n = data.length;
        int[] arr = Arrays.copyOf(data, n); // Copia del array original
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                 
                    
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        return arr;
    }
    }
    

File: EXAM/maza/u3/ExamU3/model/BubbleSort.js
export class BubbleSort {
    static sort(arr) {
        let sortedArr = [...arr];
        for (let i = 0; i < sortedArr.length - 1; i++) {
            for (let j = 0; j < sortedArr.length - i - 1; j++) {
                if (sortedArr[j] > sortedArr[j + 1]) {
                    [sortedArr[j], sortedArr[j + 1]] = [sortedArr[j + 1], sortedArr[j]];
                }
            }
        }
        return sortedArr;
    }
}
